var/spack/repos/builtin/packages/catch2/package.py

```diff
@@ -104,6 +105,9 @@ class Catch2(CMakePackage):
     version("1.3.5", sha256="f15730d81b4173fb860ce3561768de7d41bbefb67dc031d7d1f5ae2c07f0a472")
     version("1.3.0", sha256="245f6ee73e2fea66311afa1da59e5087ddab8b37ce64994ad88506e8af28c6ac")
 
+    variant("pic", when="@3:", default=True, description="Build with position-independent code")
```

Comment:
I would do:
```suggestion
    variant("pic", when="@3: ~shared", default=True, description="Build with position-independent code")
```
to avoid `+shared~pic` as a possibility.

Reply:
Wouldn't `+shared~pic` still be allowed with this change?

Reply:
Fixed 👍